### ISSUE TITLE
yarn 으로 변경하는 것을 제안드립니다!

### ISSUE THREAD

**Issue by minsoo-web** [sgsg9447/react-check-in-out-calendar#42]
지금 dependencies 관리가 잘 안 되고 있는 것 같습니다.

<img width="331" alt="image" src="https://github.com/sgsg9447/react-check-in-out-calendar/assets/57122180/cc237033-7787-40fb-a390-edb0f69f3e9a">

예시로, styled-components 사용하면 유령 의존성으로 react-is가 딸려오는데,
이런 의존성 문제를 해결하는 가장 쉬운 방식이 yarn 을 활용한 방식이 있습니다.

괜찮으시다면 이 부분도 적용해보고 싶습니다!

**Comment by sgsg9447** [comment]
네 좋습니다! 여쭤보고싶은것이, 기존의 npm 에서 yarn으로 변경한다면 설정부분만 변경해주면 되는것일까요 ? 
의존성 문제를 제외하고도 yarn을 사용하면 어떤부분이 이점이 있는지 궁금합니다.
지금까지느 사용성에서는 큰차이는 못느꼈어서 타 프로젝트에서도 npm 과 yarn을 번갈아가며 사용했던것 같습니다.
민수님의 생각이 궁금합니다!

**Comment by minsoo-web** [comment]
안녕하세요 슬기님!

우선 문의 주신 내용 답변 먼저 드리겠습니다!

> 설정부분만 변경해주면 되는것일까요 ?

yarn berry 버전을 쓴다고 명시하는 작업이 필요한데, package.json 부분이 변경되고, package-lock.json 이 yarn.lock 파일로 변경되는 등의 작업이 필요합니다.

> 의존성 문제를 제외하고도 yarn을 사용하면 어떤부분이 이점이 있는지 궁금합니다.

의존성 문제를 해결해주는 게 우선 가장 큰 이점입니다.
유령 의존성 등 불필요하게 큰 node_modules 관리가 불필요해질 때 yarn 을 통해 유령 의존성 등의 의존성 문제를 알아서 해결해주는 이점이 있습니다. 

추가로, zero-install 을 사용하면 협업에도 이점이 있습니다. 패키지 변경사항이 있을 때마다 install 을 하여 패키지 변경사항을 로컬에서 관리해야 하지만, zero install 을 적용하게 되면 zip 파일로 의존성 파일을 함께 clone 받기 때문에 이런 불편함을 개선할 수 있습니다.

또 하나의 장점은 추후에 모노레포를 도입하게 될 경우에, yarn을 통해 수월하게 모노레포 구조를 적용할 수 있다는 점도 이점입니다.
